stdlib/src/builtin/int.mojo

```diff
@@ -648,6 +648,55 @@ struct Int(
             n >>= 1
         return res
 
+    @always_inline("nodebug")
+    fn __pow__(self, exp: Self, mod: Self) -> Self:
+        """Returns `self` raised to `exp` power, modulo `mod`.
+
+        This method uses the square-and-multiply algorithm to compute
+        modular exponentiation.
+
+        Args:
+            exp: The exponent. Can be positive or negative.
+            mod: The modulus. Must be greater than 1.
+
+        Returns:
+            The result of `self` raised to `exp` power mod `mod`.
+
+        Note:
+            - If mod <= 1, returns 0 to indicate an error.
+            - For negative exponents, it computes the modular multiplicative
+              inverse of self before exponentiation. If no inverse exists,
+              returns 0.
+        """
+        if mod == 0:
+            # Modulus must be greater than 1.
+            # This should raise an exception.
```

Comment:
Hi, @JoeLoser sorry to pull you in here, but I think this falls into the same category as issue #2855 about undefined math operations leading to undefined behavior. I think we should define a standard way to deal with these soon or we might end up with a bunch of these edge cases hidden away.

Reply:
+1, but at a minimum we should add a `debug_assert` here for now.

Reply:
Thank you for the review Martin and Laszlo :) I was actually busy writing a discussion regarding this, as this seems to be a lingering around the community. 

I am too for exception raises when it comes to undefined behaviour. Joe mentioned parametric raises in one of your PRs or issues, Martin, which I believe would be a strategy. This also seems to allow oppurtunity for good documentation and a standard to follow. 

I had been following the style of `__pow__(self, exp: Self)`, of which returns zero on exception. Shall I go ahead with the `debug_assert`? Both here and on the coprime issue, for now? Or should this be put on hold until a discussion?

Reply:
It's unclear when we will have parametric raises, and until then, the precedent dictates that we prioritize performance for these cases, and use `debug_assert`. So yes, please go ahead, but also feel free to start/contribute to discussion regarding this, we love having community discussions!